BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
+{
+    private const string MainCurrenciesUrl = "https://api.cnb.cz/cnbapi/exrates/daily";
+
+    public IEnumerable<Rate> GetExchangeRates(List<string> currencies)
+    {
+        HttpClient client = new HttpClient();
```

Comment:
Every time we call GetExchangeRates, HttpClient creates new instance. Which is not efficient. We can create one instance when we run the application and we can use it everywhere we can. So how to to this? 

Add following code, line 5 in program.cs:

builder.Services.AddHttpClient<ExchangeRateService>();

You can use it with injecting in the service like:

 private readonly HttpClient _httpClient;

    public ExchangeRateService(HttpClient httpClient)
    {
        _httpClient = httpClient;
    }

Reply:
public class ExchangeRateService : IExchangeRateService
{
  private readonly string _cnbCurrenciesUrl;
  private readonly HttpClient _httpClient;

    public ExchangeRateService(IConfiguration configuration, HttpClient httpClient)
    {
        _cnbCurrenciesUrl = configuration["ExternalUrls:CnbCurrenciesUrl"];
        _httpClient = httpClient;
    }
  public async Task<IEnumerable<Rate>> GetExchangeRatesAsync(List<string> currencies){
  
  // service implementation
  }
}

Reply:
Add try-catch blocks to see if something wrong is happening. We always want to is if we have correct response.

```
 public async Task<IEnumerable<Rate>> GetExchangeRatesAsync(List<string> currencies){
  try{
  // service implementation
  }
   catch (HttpRequestException httpEx)
    {
        // Handle network-related errors (e.g., server unavailable, bad response, etc.)
        Console.WriteLine($"Network error occurred: {httpEx.Message}");
        throw new Exception("There was a problem with the network request.", httpEx);
    }
}
```

Reply:
Add try- catch inside the GetExchangeRatesAsync. We should be sure there is no error before proceed.